### ISSUE TITLE
Problem with USB audio dropouts

### ISSUE THREAD

**Issue by Rogerthat12** [zearp/Nucintosh#43]
Hi,

I have problem with dropouts and static ?noise on Focusrite 2i2 USB sound card.
Can you help me with it please ?

My system:
macOS Big Sur 11.3.1 (all other things works perfectly, except for drm content like apple movies,...)

**Comment by zearp** [comment]
Whilst we do load Apple's [GuC microcode](https://pikeralpha.wordpress.com/2017/01/14/does-apple-load-microcode-for-the-igpu/) in our iGPU to improve performance it doesn't fix anything related to DRM -- DRM will only work with a dedicated GPU. This is not something that can be fixed. The best you can do is try browsers that have Widevine (like Chrome), they might be able to play certain DRM protected content at 720p. I had no luck with Prime Video but I heard some people could watch Netflix, albeit in lower resolutions.

As for the audio interface, yours seems to require a driver/kext. I have a Teac interface that without Teac's kext will start to stutter and then just drop out like yours. Have you installed the Focusrite drivers? They have a [beta build](http://beta.focusrite.com/file.php?id=1385) that should work on Big Sur. Try using that and also make sure your interfaces firmware is up-to-date. 

If it happens with an up-to-date driver loaded you can check the log files for anything audio related it may help. You can also contact Focusrite and ask them how to figure out whats going on. I'd also be interested to hear if you experience the same issue on Catalina. There are some big changes under the hood in Big Sur when ti comes to drivers and also audio. Several of my gear doesn't have any Big Sur drivers yet, or they are in beta test stage only. So it's always a good thing to try in Catalina.

If you don't need or require Big Sur for certain software or things then I'd highly recommend sticking to Catalina for any audio production. It is much more mature and better supported by manufacturers, some of whom struggle to keep up with Apple relentless release schedule and big under the hood changes. macOS 12 is on the way already, there will never be an end until manufacturers simply stop bothering updating their drivers for newer macOS versions, some already did for "older" products. I won't go into a rant on how Apple is kind of ruining the professional reputation they have/had amongst musicians and audio pro's by moving so fast and doing so with some many deep changes, so I'll leave it at that lol.

Long story short, unfortunately there's not much I can do if the above suggestions don't work. The EFI has the USB ports setup and the chance that something in the config is causing your issue is slim. I use quite a few usb audio interfaces and gear with usb connectivity myself and have only experienced issues like you have when an interface needed a driver/kext but didn't have one installed or actively loaded. My Teac needs them and Elektron's Overbridge also does, all work without issue. My other more basic 2 in 2 out interfaces work without any drivers needed. Same for some usb DACs I use from time to time. Really haven't experienced any usb issues on these machines myself.

tl;dr
Update to the latest driver and firmware, double check the Focusrite driver/kext is loaded, sometimes macOS blocks the loading of extensions. You can check this by opening System Preference -> Privacy. On the general tab there might be a warning. See [this](https://support.focusrite.com/hc/en-gb/articles/115005086925-My-Clarett-Saffire-Red-Interface-is-not-appearing-on-Mac-OS-Big-Sur-Catalina-Mojave-High-Sierra) page for more info and what to look out for.

**Comment by Rogerthat12** [comment]
Thanks for your help. I will try Catalina, and I hope that solve my issues :) Focusrite control doesn't support my 2i2 because it's 2nd not 3rd gen. I have been using it with my MacBook Pro 13" 2010 flawlessly, now I switched for money reason. I will you get I'll let you know if I'll be more lucky with Catalina.

**Comment by zearp** [comment]
Maybe the interface is only working in 24bits, thats possible but I'm not very familiar with Focusrite products. If have more options on your MacBook then something is not right indeed. Does changing it help though? You may have to match it to the source, that is if your DAW is set to 96khz you set the interface to it too.

To make sure it's not a power issue you can connect the interface to a powered usb hub if you have one that is. If its a power issue you can try experimenting with the wake type power levels [here](https://github.com/zearp/Nucintosh/blob/master/EFI/OC/Kexts/USBPorts.kext/Contents/Info.plist#L41-L48). Very unlikely though, but a possible cause never the less.

**Comment by Rogerthat12** [comment]
Hi, I have Catalina now. It feels much more robust, not glitchy audio during recording (I had this issue on Big Sur also). Otherwise I've noticed much more better sound in Catalina, completely as same as my high Sierra MacBook Pro. Big Sur has worse sound I think, much more flat and distorted and little more trable, same AirPods.

Static noise is same, maybe slightly better, but I think that's the bad design of NUC, I easily turndown volume on monitors and turned up sound card volume. I happy now, thank you so much :)

**Comment by zearp** [comment]
Awesome! So it would be safe to conclude that the issue is caused by Big Sur. You can dig down on it more but it might be a bug not easily found or fixed. It could be something has changed the interface doesn't like and freaks out resulting in stuttering and glitches. It would be interesting to see if there are any log entries when it happens but for now your problem is fixed and to be honest I get a lot more work on Catalina as the interface is much more refined and doesn't requires 2-3 clicks to get things done. Big Sur introduces a lot of extra clicks in my workflow.

There are some big changes under the hood in Big Sur when it comes to audio, so it might well be that (especially) older h/w works a lot better on older versions of macOS. I know people who keep their main music rig on High Sierra due to compatibility, not just with h/w but also s/w. If you wanna use 32bit apps or VSTs you paid good money for you need to use older macOS too since Big Sur (and also Catalina I think) won't run them anymore.

The sound chip in the NUC isn't all that bad but also nothing special. I haven't noticed any difference in audio quality between macOS versions but I'm mainly using an interface or DAC/AMP. Either to get a digital signal for my monitors or to power my headphones. The headphone output on the NUC can't really handle anything above 80ohms properly. Perfectly fine for 99% of the users but not for us haha.

I run Catalina on my main NUC too, the biggest downsite is the constant nagging by Apple to upgrade to Big Sur. Back in the days we could ignore such upgrades but Apple disabled it in Catalina. The only way to properly ignore the Big Sur updates it to enrol your machine in MDM which is not a feasible solution. I'm hoping someone a lot smarter than me can figure out a way to stop the nags in proper way. There are some apps now that will detect and kill the Big Sur updater but by that time you've already downloaded 12gb.

I'll close this issue but feel free to post updates or some Big Sur logs if you ever fancy digging down on that. I am curious if there's anything at all. With my Teac there are no logs entries when it freaks out and the only fix is force loading a 7 year old old kext that funnily enough still works fine in Big Sur.
